contracts/factory/src/contract.rs

```diff
@@ -168,6 +170,45 @@ impl FactoryTrait for Factory {
             .get(&DataKey::Config)
             .expect("Factory: No multihop present in storage")
     }
+
+    fn get_lp_and_stake_info(env: Env, caller: Address) -> LpAndStakeInfo {
+        let all_lp_vec_addresses = get_lp_vec(&env);
+        let mut lp_share_info: Vec<LpShareInfo> = Vec::new(&env);
+        let mut stake_info: Vec<StakeInfo> = Vec::new(&env);
+
+        for address in all_lp_vec_addresses {
+            let response: LiquidityPoolInfo = env.invoke_contract(
+                &address,
+                &Symbol::new(&env, "query_pool_info_for_factory"),
+                Vec::new(&env),
+            );
+            // todo here we check if the lp balance is over 0
+            if response.pool_response.asset_lp_share.amount > 0 {
+                lp_share_info.push_back(LpShareInfo {
+                    owner: caller.clone(),
+                    amount: response.pool_response.asset_lp_share.amount,
+                })
+            }
```

Comment:
You iterate over all LP addresses, you query each pool for that pool's data (not the user's!) and then you check if each pool has `lp_share` amount bigger then 0... And you assign that general amount to a user. This is invalid.